test/components/checkbox.spec.tsx

```diff
@@ -193,6 +192,56 @@ describe('<Checkbox/>', () => {
         });
     });
 
+    it('Switches to focus state when focused', async () => {
+        const {driver: checkbox, waitForDom} = clientRenderer.render(
+            <CheckBox   value={true}/>
+        ).withDriver(CheckBoxTestDriver);
+
+        await waitForDom(() => {
+            checkbox.focus();
+            expect(checkbox.elementHasStylableState('focus')).to.equal(true);
+        });
+    });
+
+    it('Accepts "name" prop', async () => {
+        const {driver: checkbox, waitForDom} = clientRenderer.render(
+            <CheckBox  name="shlomi"/>
+        ).withDriver(CheckBoxTestDriver);
+
+        await waitForDom(() => {
+            expect(checkbox.nativeInput).to.have.attribute('name', 'shlomi');
+        });
+    });
+
+    it('Accepts "required" prop', async () => {
+        const {driver: checkbox, waitForDom} = clientRenderer.render(
+            <CheckBox  required/>
+        ).withDriver(CheckBoxTestDriver);
+
+        await waitForDom(() => {
+            expect(checkbox.nativeInput).to.have.property('required', true);
+        });
+    });
+
+    it('Accepts "autofocus" prop', async () => {
+        if (document.hasFocus()) {
+
+            const {driver: checkbox, waitForDom} = clientRenderer.render(
+                <CheckBox  autoFocus/>
+            ).withDriver(CheckBoxTestDriver);
+
+            await waitForDom(() => {
+                expect(document.activeElement).to.equal(checkbox.nativeInput);
+                expect(checkbox.elementHasStylableState('focus')).to.equal(true);
```

Comment:
We're missing a similar test for `:disabled` state, although we have a test which ensures that the native input receives the attribute.

Reply:
added a test